src/library/scala/collection/convert/IntAccumulator.scala

```diff
@@ -0,0 +1,497 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection.convert
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+import java.util.Spliterator
+import java.util.function.{Consumer, IntConsumer}
+import java.{lang => jl}
+
+import scala.collection.Stepper.EfficientSplit
+import scala.collection.{AnyStepper, Factory, IntStepper, SeqFactory, Stepper, StepperShape}
+
+/** A `IntAccumulator` is a low-level collection specialized for gathering
+ * elements in parallel and then joining them in order by merging them.
+ * This is a manually specialized variant of `AnyAccumulator` with no actual
+ * subclassing relationship with `AnyAccumulator`.
+ *
+ * TODO: doc why only Iterable, not IndexedSeq or such. Operations inherited by Seq are
+ * implemented based on length, which throws when more than MaxInt.
+ *
+ * TODO: doc performance characteristics.
+ */
+final class IntAccumulator
+  extends Accumulator[Int, AnyAccumulator, IntAccumulator]
```

Comment:
This works only because you override the inherited implementations of `fromSpecific` and `iterableFactory`.